src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -6913,4 +6929,46 @@ String StorageReplicatedMergeTree::getSharedDataReplica(
     return best_replica;
 }
 
+String StorageReplicatedMergeTree::findReplicaHavingPart(
```

Comment:
What's the difference from `StorageReplicatedMergeTree::findReplicaHavingPart(const String & part_name, bool active)`? Why do we need separate method?

Reply:
```c++
String StorageReplicatedMergeTree::findReplicaHavingPart(const String & part_name, bool active)
{
    auto zookeeper = getZooKeeper(); // which zookeeper get from current StorageReplicatedMergeTree
}
```


new fuction ,which is a static function 
zookeeper get from shardpath "ALTER TABLE simple2 FETCH PART '20200827_0_0_0' FROM 'zookeeper2:/clickhouse/tables/0/simple"
```c++
    zkutil::ZooKeeperPtr zookeeper;
    if (auxiliary_zookeeper_name != default_zookeeper_name)
        zookeeper = global_context.getAuxiliaryZooKeeper(auxiliary_zookeeper_name);
    else
        zookeeper = getZooKeeper();

    if (from.back() == '/')
        from.resize(from.size() - 1);

    if (fetch_part)
    {
        String part_name = partition->as<ASTLiteral &>().value.safeGet<String>();
        auto path_replia = findReplicaHavingPart(part_name, from, zookeeper);
}